### DISCUSSION TITLE
Incorrrect evaluation of cos (90 deg)

### DISCUSSION THREAD

**Discussion by meadow-tulips** [josdejong/mathjs#3040] [Q&A]
Tried evaluating cos (90 deg) in the demo notepad. 

![image](https://user-images.githubusercontent.com/19764187/228506548-ffbd5c98-6f64-4d8d-9f17-ec104a1628c6.png)

Showing some weird value. 
It should return 0

**Comment by josdejong** [comment]
This is the most asked question. It is a floating point round-off error.

See docs: https://mathjs.org/docs/datatypes/numbers.html#roundoff-errors

**Comment by nickkolok** [comment]
But how about a bunch of simplification rules? For angles of $15\cdot k$ degrees, representation with square roots always exists...

**Comment by josdejong** [comment]
It would be possible to implement checks for that, but since we're dealing with floating point numbers, such rounding-off logic will not work flawlessly and creates inconsistencies. For floating point calculations I think it is simply best to just accept the fact that there are round-off errors involved in general. The only real solution is to use symbolic computation, like extend `math.simplify` to replace `sin(90 deg)` and `sin(2pi)` with `0`.

**Comment by lulunac27a** [comment]
It's because numbers in JavaScript uses 64-bit double precision numbers, so 90 degrees is pi/2 radians, since pi is an irrational number which cannot be expressed as a fraction and pi cannot be expressed in precise binary number as double precision numbers uses binary notation.
